<!DOCTYPE html>
<html>
	<head>
		<meta charset="utf-8">
		<title>async和await</title>
	</head>
	<body>
		<script type="text/javascript">
			// async 函数
			/* async function fn() {
				
				// 返回的结果不是一个Promise 类型的对象，返回的结果就是成功的Promise对象
				// return '我是函数';
				
				// 抛出错误，返回的结果是一个失败的 promise
				// throw new Error('天呢，报错了');
				
				// 返回结果是一个promise 对象,函数的返回值就是new出来的返回的结果
				return new Promise((resolve, reject)=>{
					resolve('成功调用');
					// reject('不好意思调用失败')
				})
			}
			const result = fn();
			// console.log(result);
			result.then((resolve)=>{
				console.log(resolve)
			},(reject)=>{
				console.warn(reject)
			}) */
			
			// await 
			const p = new Promise((resolve, reject)=>{
				resolve('用户数据')
				// reject('调用失败')
			});
			async function main() {
				try{
					let result = await p;
					console.log(result);
				} catch(e) {
					console.log(e)
				}
			}
			main();
		</script>
	</body>
</html>
